pallets/apollo-platform/src/lib.rs

```diff
@@ -0,0 +1,1429 @@
+                / FixedWrapper::from(balance!(num_of_pools)))
+            .try_into_balance()
+            .unwrap_or(0);
+
+            for (asset_id, mut pool_info) in <PoolData<T>>::iter() {
+                if pool_info.is_removed {
+                    continue;
+                }
+                pool_info.basic_lending_rate = basic_lending_rate;
+                pool_info.borrowing_rewards_rate = borrowing_rewards_rate;
+                <PoolData<T>>::insert(asset_id, pool_info);
+            }
+
+            if let Some(mut pool_info) = <PoolData<T>>::get(asset_id) {
+                pool_info.basic_lending_rate = basic_lending_rate;
+                pool_info.borrowing_rewards_rate = borrowing_rewards_rate;
+                pool_info.loan_to_value = loan_to_value;
+                pool_info.liquidation_threshold = liquidation_threshold;
+                pool_info.optimal_utilization_rate = optimal_utilization_rate;
+                pool_info.base_rate = base_rate;
+                pool_info.slope_rate_1 = slope_rate_1;
+                pool_info.slope_rate_2 = slope_rate_2;
+                pool_info.reserve_factor = reserve_factor;
+                pool_info.is_removed = false;
+                <PoolData<T>>::insert(asset_id, pool_info);
+            } else {
+                // Create a new pool
+                let new_pool_info = PoolInfo {
+                    total_liquidity: 0,
+                    total_borrowed: 0,
+                    total_collateral: 0,
+                    basic_lending_rate,
+                    profit_lending_rate: 0,
+                    borrowing_rate: 0,
+                    borrowing_rewards_rate,
+                    loan_to_value,
+                    liquidation_threshold,
+                    optimal_utilization_rate,
+                    base_rate,
+                    slope_rate_1,
+                    slope_rate_2,
+                    reserve_factor,
+                    rewards: 0,
+                    is_removed: false,
+                };
+
+                <PoolData<T>>::insert(asset_id, new_pool_info);
+
+                // Add pool to PoolsByBlock map
+                let num_of_pools = <PoolsByBlock<T>>::iter().count() as u32;
+                let block_number: BlockNumberFor<T> = num_of_pools.into();
+                <PoolsByBlock<T>>::insert(block_number, asset_id);
+            }
+
+            Self::deposit_event(Event::PoolAdded(user, asset_id));
+            Ok(().into())
+        }
+
+        /// Lend token
+        #[pallet::call_index(1)]
+        #[pallet::weight(<T as Config>::WeightInfo::lend())]
+        pub fn lend(
+            origin: OriginFor<T>,
+            lending_asset: AssetIdOf<T>,
+            lending_amount: Balance,
+        ) -> DispatchResultWithPostInfo {
+            let user = ensure_signed(origin)?;
+
+            ensure!(lending_amount > 0, Error::<T>::InvalidLendingAmount);
```

Comment:
Please, check that `lending_amount` is more than some minimal amount to prevent attack with blowing of `UserLendingInfo`.